File: projects/OMSATrackingAPI/OMSATrackingAPI.DAL/Models/Bus.cs
namespace OMSATrackingAPI.DAL.Models
{
    public class Bus : BaseEntity<int>
    { 
        public string Name { get; set; } = string.Empty;
        public string Latitude { get; set; } = string.Empty;
        public string Longitude { get; set; } = string.Empty;
        public string Plate { get; set; } = string.Empty;
        public string EstimatedArrivalHour { get; set; } = string.Empty;
        public int PassengerLimit { get; set; }
        public int RouteId { get; set; }
        public int StopId { get; set; }

        // Navigation properties
        public Route Route { get; set; } = null!;
        public BusStop Stop { get; set; } = null!;
        public FavoriteRoute FavoriteRoute { get; set; } = null!;
        public Driver Driver { get; set; } = null!;
    }
}

File: projects/omsa_tracking_app/lib/models/bus.dart
import 'package:omsa_tracking_app/models/driver.dart';

class Bus {
  int id;
  String name;
  double latitude;
  double longitude;
  String plate;
  String estimatedArrivalHour;
  int passengerLimit;
  int routeId;
  Driver driver;

  Bus({
    required this.id,
    required this.name,
    required this.latitude,
    required this.longitude,
    required this.plate,
    required this.estimatedArrivalHour,
    required this.passengerLimit,
    required this.routeId,
    required this.driver,
  });

  factory Bus.fromJson(Map<String, dynamic> json) {
    return Bus(
      id: json['id'],
      name: json['name'],
      latitude: double.parse(json['latitude']),
      longitude: double.parse(json['longitude']),
      plate: json['plate'],
      estimatedArrivalHour: json['estimatedArrivalHour'],
      passengerLimit: json['passengerLimit'],
      routeId: json['routeId'],
      driver: Driver.fromJson(json['driver']),
    );
  }
}
